src/Reducer.js

```diff
@@ -163,7 +163,18 @@ function inject(state, action, props, scenes) {
     case ActionConst.JUMP:
       assert(state.tabs, `Parent=${state.key} is not tab bar, jump action is not valid`);
       ind = -1;
-      state.children.forEach((c, i) => { if (c.sceneKey === action.key) { ind = i; } });
+      state.children.forEach((child, i) => {
+        if (child.sceneKey === action.key) {
+          ind = i;
+
+          // Pass the props to the wrapped tab scene
+          if (props.children) delete props.children;
+          const key = child.children[0].sceneKey;
+          const wrappedScene = scenes[key];
+          child.children = [{ ...props, ...wrappedScene }];
+        }
+      });
+
       assert(ind !== -1, `Cannot find route with key=${action.key} for parent=${state.key}`);
```

Comment:
Good looking out. It just seemed like the way to go with all of the shallow slices pointing back to previous state objects, and direct setting of state vars [here](https://github.com/aksonov/react-native-router-flux/blob/master/src/Reducer.js#L178) and [here](https://github.com/aksonov/react-native-router-flux/blob/master/src/Reducer.js#L191).

This does give me a change to better cover edge cases though. I noticed when nesting tabs things stop behaving as expected. I'm thinking at least tests for double nested tab routes should cover it. I'll take another stab at it this weekend.